lib/dataset/shape.cpp

```diff
@@ -117,10 +117,13 @@ DataArray concat(const scipp::span<const DataArray> das, const Dim dim) {
   const auto &coords = map(das, get_coords);
   for (auto &&[d, coord] : concat_maps(map(das, get_meta), dim)) {
     if (d == dim || std::any_of(coords.begin(), coords.end(),
-                                [&d = d](auto &_) { return _.contains(d); }))
+                                [&d = d](auto &_) { return _.contains(d); })) {
+      coord.set_aligned(true);
       out.coords().set(d, std::move(coord));
-    else
+    } else {
+      coord.set_aligned(false);
       out.attrs().set(d, std::move(coord));
```

Comment:
Shouldn't you set it as an attr, i.e., the reverse of what slicing does? An then the branching can be replaced by just setting the flag.

Reply:
Is does get set as an attr. Do you mean it should be a coord instead?

Reply:
Yes.